holoviews/plotting/bokeh/element.py

```diff
@@ -286,6 +286,17 @@ def _hover_opts(self, element):
         dims += element.dimensions()
         return list(util.unique_iterator(dims)), {}
 
+    def _add_element_group_and_label_hover_tooltips(self, element, tooltips):
+        # return group and label if present
+        group_label = []
+        # group is the element type by default (e.g. Curve)
+        # skip if 'group' and 'label' are data dims that are already in tooltips
+        if hasattr(element, 'group') and element.group != element.param.group.default and 'group' not in [t[0].lower() for t in tooltips]:
+            group_label.append(('Group', element.group))
+        if hasattr(element, 'label') and element.label and 'label' not in [t[0].lower() for t in tooltips]:
+            group_label.append(('Label', element.label))
+        return group_label + tooltips
```

Comment:
```suggestion
    def _add_element_group_and_label_hover_tooltips(self, element, tooltips):
       """ 
       Add elements group and label to the hover tooltip if they have changed from default.
       Will not add elements group or label if it is part of data dimension
       """
        if hasattr(element, 'group') and element.group != element.param.group.default and not any('group' == t[0].lower() for t in tooltips):
            tooltips.append(('Group', element.group))
        if hasattr(element, 'label') and element.label and not any('label' == t[0].lower() for t in tooltips):
            tooltips.append(('Label', element.label))
```

This change does 3 things:
1) Make comment docstring. This will make it show up with an LSP.
2) Add to the existing tooltip instead of creating a new tooltip (this will also need a change in the next function) as we don't return anything anymore
3) Exit early if 'group' or 'label' are present in the tooltips.

What do you think about the changes?

(This is written as a suggestion, so please double-check the suggestion)

Reply:
Thanks for the guidance re: not any(...) to end early, and generally about mutating in place if possible. That's all very helpful.

Me thinks Group and Label should be displayed first in the hover tooltips, which is why I was prepending rather than appending. The hierarchy being: 1. Groups of Elements 2. Label of an entire element 2. Names of an element's individual data dims. So how about the following to achieve in-place prepending, I just pushed this:

```python
def _add_element_group_and_label_hover_tooltips(self, element, tooltips):
       """ 
       Prepend the element's group and label to the hover tooltip if they have changed from default.
       Will not prepend element's group or label if 'group' or 'label' exist as data dimensions
       """
        if hasattr(element, 'label') and element.label and not any('label' == t[0].lower() for t in tooltips):
            tooltips.insert(0, ('Label', element.label))
        if hasattr(element, 'group') and element.group != element.param.group.default and not any('group' == t[0].lower() for t in tooltips):
            tooltips.insert(0, ('Group', element.group))
```